tests/providers/amazon/aws/operators/test_eks.py

```diff
@@ -0,0 +1,299 @@
+    ):
+        nodegroup_names: List[str] = NAME_LIST
+        cluster_name: str = random_names()
+        response = dict(cluster=cluster_name, nodegroups=nodegroup_names, nextToken=DEFAULT_NEXT_TOKEN)
+        mock_describe_nodegroup.return_value = EMPTY_NODEGROUP
+        mock_list_nodegroups.return_value = response
+
+        self.describe_all_nodegroups_operator.execute({})
+
+        mock_list_nodegroups.assert_called_once()
+        assert mock_describe_nodegroup.call_count == len(nodegroup_names)
+
+    @mock.patch.object(EKSHook, "list_nodegroups")
+    @mock.patch.object(EKSHook, "describe_nodegroup")
+    def test_no_nodegroups_exist(self, mock_describe_nodegroup, mock_list_nodegroups):
+        mock_list_nodegroups.return_value = dict(nodegroups=list(), token="")
+
+        self.describe_all_nodegroups_operator.execute({})
+
+        mock_list_nodegroups.assert_called_once()
+        mock_describe_nodegroup.assert_not_called()
+
+
+class TestEKSDescribeClusterOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_name: str = random_names()
+
+        self.describe_cluster_operator = EKSDescribeClusterOperator(
+            task_id=TASK_ID, cluster_name=self.cluster_name
+        )
+
+    @mock.patch.object(EKSHook, "describe_cluster")
+    def test_describe_cluster(self, mock_describe_cluster):
+        mock_describe_cluster.return_value = DESCRIBE_CLUSTER_RESULT
+
+        self.describe_cluster_operator.execute({})
+
+        mock_describe_cluster.assert_called_once_with(name=self.cluster_name, verbose=False)
+
+
+class TestEKSDescribeNodegroupOperator(unittest.TestCase):
+    def setUp(self):
+        self.cluster_name = random_names()
+        self.nodegroup_name = random_names()
+
+        self.describe_nodegroup_operator = EKSDescribeNodegroupOperator(
+            task_id=TASK_ID, cluster_name=self.cluster_name, nodegroup_name=self.nodegroup_name
+        )
+
+    @mock.patch.object(EKSHook, "describe_nodegroup")
+    def test_describe_nodegroup(self, mock_describe_nodegroup):
+        mock_describe_nodegroup.return_value = DESCRIBE_NODEGROUP_RESULT
+
+        self.describe_nodegroup_operator.execute({})
+
+        mock_describe_nodegroup.assert_called_once_with(
+            clusterName=self.cluster_name, nodegroupName=self.nodegroup_name, verbose=False
+        )
+
+
+class TestEKSListClustersOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_names: List[str] = NAME_LIST
+
+        self.list_clusters_operator = EKSListClustersOperator(task_id=TASK_ID)
+
+    @mock.patch.object(EKSHook, "list_clusters")
+    def test_list_clusters(self, mock_list_clusters):
+        mock_list_clusters.return_value = self.cluster_names
+
+        self.list_clusters_operator.execute({})
+
+        mock_list_clusters.assert_called_once()
+
+
+class TestEKSListNodegroupsOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_name: str = random_names()
```

Comment:
Is it needed? Why can't you use a constant value?

Reply:
Removed in upcoming revision.

Reply:
Corrected in https://github.com/apache/airflow/pull/16571/commits/76c7ad635948191764a323a9be03fa965fc91ff5